### PR TITLE
Training reproducibility

### PR DESCRIPTION
**Pull request by cbovalo**
This PR aims at finding the configuration to reproduce early acceptable experiments. It adresses issue #159.

What changed in this PR:

* added the keyword `obs_step_btw_t0` in order to subsample the validation/test period,
* changed the learning rate scheduler to reuse the former one (from Hugging Face's `transformers`).

Be carefull as now the learning rate and others parameters for the optimizer are in the model config file.

### CODE REVIEW

File: py4cast/lightning.py [hunk 1]
```diff
@@ -282,7 +291,6 @@ def __init__(
     #############################################################
 
     def setup(self, stage=None):
-        self.logger.log_hyperparams(self.hparams, metrics={"val_mean_loss": 0.0})
```
[the remark this answers is not in the document]
Comment (cbovalo) [reply py4cast/lightning.py:285]: Because you artificially add a point to the `val_mean_loss` curve. There is no reason the validation loss equals 0 at epoch 0.

**Review by kalharko — APPROVED** [review]

**Review by LBerth — APPROVED** [review]
